Tests/SwiftSyntaxTest/SyntaxConvenienceMethodsTests.swift

```diff
@@ -0,0 +1,32 @@
+import XCTest
+import SwiftSyntax
+
+public class SyntaxConvenienceMethodsTests: XCTestCase {
+
+  public func testFloatingValues() {
+    testFloatingValue(text: "5.3_8", expectedValue: 5.38)
+    testFloatingValue(text: "12e3", expectedValue: 12000.0)
+    testFloatingValue(text: "32E1", expectedValue: 320.0)
+    testFloatingValue(text: "0xdEFACE.C0FFEEp+1", expectedValue: 29226397.507810354)
+    testFloatingValue(text: "0xaffab1e.e1fP-2", expectedValue: 46131911.72064209)
+  }
+
+  private func testFloatingValue(text: String, expectedValue: Double) {
+    let digits = SyntaxFactory.makeFloatingLiteral(text)
+    let literalExpr = SyntaxFactory.makeFloatLiteralExpr(floatingDigits: digits)
+    XCTAssertEqual(literalExpr.floatingValue, expectedValue)
+  }
```

Comment:
Just a stylistic request, could you move this `testFloatingValue` and `testIntegerValue` out of `SyntaxConvenienceMethodsTests` type? They are not "test" methods for XCTest and can be a free functions.

Reply:
Done.